servers/navigation_server_2d.cpp

```diff
@@ -335,7 +335,29 @@ real_t FORWARD_1_C(map_get_edge_connection_margin, RID, p_map, rid_to_rid);
 void FORWARD_2_C(map_set_link_connection_radius, RID, p_map, real_t, p_connection_radius, rid_to_rid, real_to_real);
 real_t FORWARD_1_C(map_get_link_connection_radius, RID, p_map, rid_to_rid);
 
-Vector<Vector2> FORWARD_5_R_C(vector_v3_to_v2, map_get_path, RID, p_map, Vector2, p_origin, Vector2, p_destination, bool, p_optimize, uint32_t, p_layers, rid_to_rid, v2_to_v3, v2_to_v3, bool_to_bool, uint32_to_uint32);
+Ref<NavigationPathQueryResult2D> NavigationServer2D::map_query_path(RID p_map, const Ref<NavigationPathQueryParameters2D> &p_parameters) const {
```

Comment:
Are there any implications of using the non-macrod version of the function? I'm not super familiar with the server interface here.

Reply:
Not really.  The macros generate these functions in cases where the conversion is trivial.  In this case, since there is more work to do, I thought implementing the method myself was warranted (also helped with debugging).  If you look around the class, you'll see a few other places where the macros are not used.